doc/ref_arch/kubernetes/chapters/chapter05.md

```diff
@@ -68,30 +94,159 @@ between tenants
+often recommended to isolate the etcd servers behind a firewall that only the
+API servers may access.
+    
+### 5.4.6 Controlling access to the Kubelet
+Kubelets expose HTTPS endpoints which grant powerful control over the node and
+containers. By default Kubelets allow unauthenticated access to this API.
+Production clusters should enable Kubelet authentication and authorization
+    
+### 5.4.7 Securing Kubernetes Dashboard 
+The Kubernetes dashboard is a webapp for managing your cluster. It it is not a
+part of the Kubernetes cluster itself, it has to be installed by the owners of
+the cluster. Thus, there are a lot of tutorials on how to do this.
+Unfortunately, most of them create a service account with very high privileges.
+This caused Tesla and some others to be hacked via such a poorly configured K8s
+dashboard. (Reference: Tesla cloud resources are hacked to run
+cryptocurrency-mining malware -
+https://arstechnica.com/information-technology/2018/02/tesla-cloud-resources-are-hacked-to-run-cryptocurrency-mining-malware/)
+
+To prevent attacks via the dashboard, you should follow some tips:
+    
+* Do not expose the dashboard without additional authentication to the public.
+  There is no need to access such a powerful tool from outside your LAN
+* Turn on RBAC, so you can limit the service account the dashboard uses
+* Do not grant the service account of the dashboard high privileges
+* Grant permissions per user, so each user only can see what he is supposed to
+  see
+* If you are using network policies, you can block requests to the dashboard
+  even from internal pods (this will not affect the proxy tunnel via kubectl
+  proxy)
+* Before version 1.8, the dashboard had a service account with full privileges,
+  so check that there is no role binding for cluster-admin left.
+* Deploy the dashboard with an authenticating reverse proxy, with multi-factor
+  authentication enabled. This can be done with either embeded OIDC id_tokens or
+  using Kubernetes Impersonation. This allows you to use the dashboard with the
+  user's credentials instead of using a privileged ServiceAccount. This method
+  can be used on both on-prem and managed cloud clusters.
 
 ## 5.5 Use Namespaces to Establish Security Boundaries
 Namespaces in Kubernetes is the first level of isolation between components. It
 is easier to apply security controls (Network Policies, Pod policies, etc.) to
 different types of workloads when deployed in separate Namespaces.
 
 ## 5.6 Separate Sensitive Workload
-To limit the potential impact of a compromise, it is recommended to run sensitive
-workloads on a dedicated set of machines. This approach reduces the risk of a
-sensitive application being accessed through a less-secure application that
-shares a container runtime or host.
+To limit the potential impact of a compromise, it is recommended to run
+sensitive workloads on a dedicated set of machines. This approach reduces the
```

Comment:
I think we use the term "nodes" not "machines"

Reply:
Done